### PR TITLE
added correct handling of job success. Added covering tests to the Jo…

### PR DESCRIPTION
**Pull request by davehowlett**
…bContextMessageHandler and an acceptance test for future use

### CODE REVIEW

File: src/SFA.DAS.Payments.EarningEvents.Application/Handlers/JobContextMessageHandler.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ private async Task<bool> HandleSubmissionEvents(JobContextMessage message)
         {
             if (message.Topics.Any())
             {
-                var subscriptionMessage = message.Topics.SingleOrDefault(m => m.SubscriptionName == "GenerateFM36Payments");
+                var subscriptionMessage = message.Topics[message.TopicPointer];
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.EarningEvents.Application/Handlers/JobContextMessageHandler.cs]: I think we should check if the Topic Count matches the Topic Pointer number so we don't out of index exception

File: src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj [hunk 2]
```diff
@@ -90,6 +90,7 @@
     <Compile Include="Properties\AssemblyInfo.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (carlbm) [inline src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj]: Can you remove this?

File: src/SFA.DAS.Payments.AcceptanceTests.Core/SFA.DAS.Payments.AcceptanceTests.Core.csproj [hunk 3]
```diff
@@ -418,6 +418,7 @@
     <Compile Include="Services\UkprnService.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
```
Comment (carlbm) [inline src/SFA.DAS.Payments.AcceptanceTests.Core/SFA.DAS.Payments.AcceptanceTests.Core.csproj]: Can you remove this?

File: src/SFA.DAS.Payments.EarningEvents.Application/Handlers/JobContextMessageHandler.cs (hunk 1 above)
Comment (davehowlett) [reply src/SFA.DAS.Payments.EarningEvents.Application/Handlers/JobContextMessageHandler.cs]: changed and tested

File: src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj (hunk 2 above)
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.Components.Core/SFA.DAS.Payments.AcceptanceTests.Components.Core.csproj]: removed

File: src/SFA.DAS.Payments.AcceptanceTests.Core/SFA.DAS.Payments.AcceptanceTests.Core.csproj (hunk 3 above)
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.Core/SFA.DAS.Payments.AcceptanceTests.Core.csproj]: removed

**Pull request merged**